### PR TITLE
physics material

### PR DESCRIPTION
**Pull request by YunHsiao**
* add support for physics materials in cannon
* tests-3d skiing test scene for several materials
* minor fixes:
  * many linter error
  * more natural defines about RGBE pngs

### CODE REVIEW

File: examples/tests-3d/files.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-var files = {
+window.files = {
```
Comment (dabingnn) [inline examples/tests-3d/files.js]: what is the advantage of changing it to window.files?

File: examples/tests-3d/loadScene.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // load level
-function load(baseUrl, sceneName, onLevelLoaded) {
-  cc.resl({
+window.load = function (baseUrl, sceneName, onLevelLoaded) {
```
Comment (dabingnn) [inline examples/tests-3d/loadScene.js]: same as here

File: examples/tests-3d/files.js (hunk 1 above)
Comment (YunHsiao) [reply examples/tests-3d/files.js]: to fix some awkward linter errors...

File: lib/framework/physics/cannon-body.js [hunk 3]
```diff
@@ -24,9 +24,10 @@ export default class CannonBody extends CANNON.Body {
       // TODO: uniform event interface across engines
       this._entity.emit('collide', event);
     });
-    this.updateIn = this.updateIn.bind(this);
-    this.updateOut = this.updateOut.bind(this);
-    this.updateIn(); // TODO: emit event after transform change perhaps?
+    this._cancelOutGravity = this._cancelOutGravity.bind(this);
+    this._updateIn = this._updateIn.bind(this);
```
Comment (dabingnn) [inline lib/framework/physics/cannon-body.js]: arrow function could be used here to make it more elegant. 
```javascript
this._updateIn = ()=> {....}
```

**Pull request merged**